packages/sdk-metrics/src/aggregator/ExponentialHistogram.ts

```diff
@@ -0,0 +1,606 @@
+  };
+  hasMinMax: boolean;
+  min: number;
+  max: number;
+  sum: number;
+  count: number;
+  scale: number;
+  zeroCount: number;
+}
+
+// HighLow is a utility class used for computing a common scale for
+// two exponential histogram accumulations
+class HighLow {
+  static combine(h1: HighLow, h2: HighLow): HighLow {
+    return new HighLow(Math.min(h1.low, h2.low), Math.max(h1.high, h2.high));
+  }
+  constructor(public low: number, public high: number) {}
+}
+
+const MAX_SCALE = 20;
+const DEFAULT_MAX_SIZE = 160;
+const MIN_MAX_SIZE = 2;
+
+export class ExponentialHistogramAccumulation implements Accumulation {
+  constructor(
+    public startTime: HrTime = startTime,
+    private _maxSize = DEFAULT_MAX_SIZE,
+    private _recordMinMax = true,
+    private _sum = 0,
+    private _count = 0,
+    private _zeroCount = 0,
+    private _min = Number.POSITIVE_INFINITY,
+    private _max = Number.NEGATIVE_INFINITY,
+    private _positive = new Buckets(),
+    private _negative = new Buckets(),
+    private _mapping: Mapping = getMapping(MAX_SCALE)
+  ) {
+    if (this._maxSize < MIN_MAX_SIZE) {
+      diag.warn(`Exponential Histogram Max Size set to ${this._maxSize}, \
+                changing to the minimum size of: ${MIN_MAX_SIZE}`);
+      this._maxSize = MIN_MAX_SIZE;
+    }
+  }
+
+  /**
+   * record updates a histogram with a single count
+   * @param {Number} value
+   */
+  record(value: number) {
+    this.updateByIncrement(value, 1);
+  }
+
+  /**
+   * Sets the start time for this accumulation
+   * @param {HrTime} startTime
+   */
+  setStartTime(startTime: HrTime): void {
+    this.startTime = startTime;
+  }
+
+  /**
+   * Returns the datapoint representation of this accumulation
+   * @param {HrTime} startTime
+   */
+  toPointValue(): InternalHistogram {
+    return {
+      hasMinMax: this._recordMinMax,
+      min: this.min(),
+      max: this.max(),
+      sum: this.sum(),
+      positive: {
+        offset: this.positive().offset(),
+        bucketCounts: this.positive().counts(),
+      },
+      negative: {
+        offset: this.negative().offset(),
+        bucketCounts: this.negative().counts(),
+      },
+      count: this.count(),
+      scale: this.scale(),
+      zeroCount: this.zeroCount(),
+    };
+  }
+
+  /**
+   * @returns {Number} The sum of values recorded by this accumulation
+   */
+  sum(): number {
+    return this._sum;
+  }
+
+  /**
+   * @returns {Number} The minimum value recorded by this accumulation
+   */
+  min(): number {
+    return this._min;
+  }
+
+  /**
+   * @returns {Number} The maximum value recorded by this accumulation
+   */
+  max(): number {
+    return this._max;
+  }
+
+  /**
+   * @returns {Number} The count of values recorded by this accumulation
+   */
+  count(): number {
+    return this._count;
+  }
+
+  /**
+   * @returns {Number} The number of 0 values recorded by this accumulation
+   */
+  zeroCount(): number {
+    return this._zeroCount;
+  }
```

Comment:
Seems like this is making these private properties available as read only values? The `get` keyword allows you to define getters for this avoiding the need for `fn()` syntax at call time

Reply:
I changed these. For consistency I changed a few others (in Buckets and Mapping). Let me know if I've gone too far, some of them are getters in-spirit.